### ISSUE TITLE
Document that some specific properties may not be removed when using kubectl apply

### ISSUE THREAD

**Issue by stephen-dexda** [kubernetes/kubernetes#82525]
Per #72519, serviceAccountName (and serviceAccount) are not removed when they are deleted from a resource spec. which is then passed to `kubectl apply`.

I hit this exact bug, and there is nothing in the docs at https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#kubectl-apply to warn about the existence of special cases like that.

If there is no obvious fix that would address non-removal of special cases, then this caveat should be mentioned.

**Comment by stephen-dexda** [comment]
@kubernetes/sig-cli
@kubernetes/sig-auth

**Comment by stephen-dexda** [comment]
@kubernetes/sig-cli-bugs
@kubernetes/sig-auth-bugs

**Comment by liggitt** [comment]
/remove-sig auth
/remove-kind bug
/kind documentation

Defaulted fields cannot be unset without getting re-defaulted. This is not specific to apply, but is reasonable to mention in documentation.

**Comment by stephen-dexda** [comment]
> Fields that are represented in multiple places in the same object (like Pod spec.serviceAccount and spec.serviceAccountName) must both be updated or explicitly unset.

I'd consider this behaviourally a bug, since the user can't know this without digging into the history/provenance of every property, and further because the state of the resource passed to `apply` currently depends on the previous state of the resource. A user would reasonably expect that the result of `apply` would be the same whether the resource was being created or updated.

**Comment by roy-work** [comment]
To me, during a `kubectl apply`, I'm _setting_ the state of the object, I'm not trying to patch it. (I'd be calling `kubectl patch` if that were the case.) To have fields on the previous object survive into the new object like this is _incredibly_ surprising behavior.

**Comment by ncc1701j** [comment]
> To me, during a `kubectl apply`, I'm _setting_ the state of the object¹, I'm not trying to patch it. (I'd be calling `kubectl patch` if that were the case.) To have fields on the previous object survive into the new object like this is _incredibly_ surprising behavior.
> 
> I don't think this is an issue with the docs; this is a bug in the current behavior of `kubectl apply`. What makes those two fields special?
> 
> ¹admittedly, there are some status fields, and things like `resourceVersion`; but those are metadata that are at least the sort of thing one expects. `serviceAccount` is not.

I agree, this really feels like a bug. It even makes the rollback fail, when serviceAccount is a field that is been added (upgrade) / removed (rollback).
